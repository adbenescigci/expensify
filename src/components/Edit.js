import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense } from '../actions/expenses';
import {removeExpense} from '../actions/expenses';

export class EditExpensePage extends React.Component {

    onEdit =  (expense)=>{
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }

    onRemove = () => {
        this.props.removeExpense({id: this.props.expense.id });
        this.props.history.push('/');
    }

    render () {
        return (
            <div>
                <ExpenseForm 
                    expense = {this.props.expense}
                    onSubmit = {this.onEdit}/>  
                <button onClick = {this.onRemove} > Remove </button>   
            </div>
            )
    }
}
const mapDispatchToProps = (dispatch, props) => ({
    editExpense: (id,expense)=>{
       dispatch(editExpense(id, expense))
    },
    removeExpense : (data) => {
       dispatch(removeExpense(data))
    }
})

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense)=>{
            return expense.id === props.match.params.id.substring(1)
        })
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);