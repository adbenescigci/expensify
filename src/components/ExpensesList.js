import React from 'react';
import { connect } from 'react-redux';
import ExpensesListItems from './ExpensesListItems';
import Selection from '../selectors/expenses'

export const ExpensesList = (props) => (
    <div>
        <h1> ExpensesList </h1>
      {
        props.expenses.length === 0 ? (
              <p>There is no expenses </p>
        ) : (
            props.expenses.map((expense)=>{
                return  <ExpensesListItems key= {expense.id} {...expense}/>
          }
        ))
     }
    </div>
)

const mapStateProps = (state) => {
    return {
        expenses : Selection ( state.expenses, state.filters)
    }
};

export default connect(mapStateProps)(ExpensesList);


 