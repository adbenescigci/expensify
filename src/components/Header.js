import React from 'react';
import { NavLink } from 'react-router-dom';

export default function Header () {
    return (
        <header>
         <h1> Duzenleme</h1>
         <NavLink to='/' activeClassName='is-active' exact={true}> Ana Sayfa </NavLink>
         <NavLink to='/create' activeClassName='is-active'> Odeme Ekle </NavLink>
         <NavLink to='/help' activeClassName='is-active'> Help </NavLink>
        </header>
    )
}



/*const Header = () => (
    <header>
     <h1> Duzenleme</h1>
     <NavLink to='/' activeClassName='is-active' exact={true}> Ana Sayfa </NavLink>
     <NavLink to='/create' activeClassName='is-active'> Ekle </NavLink>
     <NavLink to='/help' activeClassName='is-active'> Help Page </NavLink>
     
    </header>
)

export default Header;*/