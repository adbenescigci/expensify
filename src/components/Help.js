import React from 'react';

const Help = () => (
    <div>
        This is from HelpPage component
    </div>
)

export default Help;