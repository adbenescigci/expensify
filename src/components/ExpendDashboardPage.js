import React from 'react';
import ExpensesList from './ExpensesList';
import ExpenseListFilters from './ExpenseListFilters' ;


const ExpendDashboardPage = () => (
    <div>
        <ExpenseListFilters/>
        <ExpensesList/>
        
    </div>
)

export default ExpendDashboardPage;