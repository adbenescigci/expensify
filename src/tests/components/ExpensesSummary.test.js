import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should test ExpensesSummary with 1 expense', ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount = {1} expensesTotal = {12}/>);
    expect(wrapper).toMatchSnapshot()

})


test('should test ExpensesSummary with  multiple expenses', ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount = {1} expensesTotal = {123245655}/>);
    expect(wrapper).toMatchSnapshot()

})