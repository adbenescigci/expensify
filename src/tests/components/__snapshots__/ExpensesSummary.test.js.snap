// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test ExpensesSummary with  multiple expenses 1`] = `
<div>
  <h1>
     Viewing 
    3
     
    expenses
     totalling 
    $1,232,456.55
     
  </h1>
</div>
`;

exports[`should test ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
     Viewing 
    1
     
    expense
     totalling 
    $0.12
     
  </h1>
</div>
`;
